packages/core/components/icon/index.js

```diff
@@ -1,50 +1,63 @@
-import React from "react";
+import React, { useContext } from "react";
 import PropTypes from "prop-types";
-import styled, { withTheme } from "styled-components";
-import { size } from "styled-system";
-import Text from "../text";
+import styled, { ThemeContext } from "styled-components";
+import { themeGet } from "styled-system";
+import { COMMON, FLEX_ITEM, LAYOUT, MISC, POSITION } from "../../constants";
 
-/* helpful links
-  SVG optimization
-  - https://github.com/svg/svgo
-  SVG Transformation or inspection
-  - https://github.com/elrumordelaluz/svgson-next
-  - https://github.com/Angelmmiguel/svgi
-*/
-const StyledIcon = styled(Text)`
-  ${size}
+const StyledIcon = styled.svg`
   vertical-align: middle;
+  font-size: inherit;
+  line-height: inherit;
+
+  ${p =>
+    !p.size &&
+    css`
+      width: 1em;
+      height: 1em;
+    `}
+
+  ${themeGet("icon.styles")};
+  ${COMMON}
+  ${FLEX_ITEM}
+  ${LAYOUT}
+  ${MISC}
+  ${POSITION}
 `;
 
-const Icon = ({ children, ...props }) => {
-  const icon = props.theme.icons[props.name];
+const Icon = ({ ...p }) => {
+  const themeContext = useContext(ThemeContext);
+  const icon = themeContext.icons[p.name];
   return (
     <StyledIcon
-      xmlns="http://www.w3.org/2000/svg"
+      aria-hidden={p.a11yHidden}
+      aria-label={p.a11yLabel}
       fill="currentColor"
-      viewBox={icon ? icon.viewBox : "0 0 24 24"}
-      {...props}
+      viewBox={icon ? icon.viewBox : p.viewBox ? p.viewBox : "0 0 24 24"}
+      xmlns="http://www.w3.org/2000/svg"
+      {...p}
     >
-      <path d={icon ? icon.path : ""} fill="currentColor" />
+      <path d={icon ? icon.path : p.path ? p.path : ""} fill="currentColor" />
     </StyledIcon>
   );
 };
 
 Icon.propTypes = {
-  ...Text.propTypes,
-  ...size.propTypes,
-  name: PropTypes.string.isRequired
+  ...COMMON.propTypes,
+  ...FLEX_ITEM.propTypes,
+  ...LAYOUT.propTypes,
+  ...MISC.propTypes,
+  ...POSITION.propTypes,
+  name: PropTypes.string,
+  path: PropTypes.string,
+  viewBox: PropTypes.string,
+  a11yHidden: PropTypes.bool,
+  a11yLabel: PropTypes.string
```

Comment:
Might make sense to change the a11y props to `ariaHidden` and `ariaLabel` to conform to standards, since they get assigned to those attributes anyway.

On further thought, you could just drop `a11yHidden`, `a11yLabel`, and even `viewBox` entirely from the propTypes, since they're standard HTML and/or SVG attributes that will just get passed to the SVG. If you do this, you could still mention in the docs page that it accepts all SVG props, etc.

Reply:
Good call. I removed them but added `role="img"` since this is expected 99% of the time for svg icons.